lib/charms/finos_legend_libs/v0/legend_operator_base.py

```diff
@@ -751,14 +751,18 @@ def _update_gitlab_relation_callback_uris(self):
             return
 
         redirect_uris = self._get_legend_gitlab_redirect_uris()
-        legend_gitlab.set_legend_gitlab_redirect_uris_in_relation_data(
-            relation.data[self.app], redirect_uris)
+        if self.unit.is_leader():
+            legend_gitlab.set_legend_gitlab_redirect_uris_in_relation_data(
+                relation.data[self.app], redirect_uris)
 
     def _on_config_changed(self, event: charm.ConfigChangedEvent):
         """Refreshes the service config."""
         super()._on_config_changed(event)
         # NOTE(claudiub): We need to update the gitlab integrator with the new URIs as well.
-        self._update_gitlab_relation_callback_uris()
+        # We only want to do anything here if we're the leader to avoid
+        # collision if we've scaled out this application.
+        if self.unit.is_leader():
```

Comment:
You're doing this check in ``_update_gitlab_relation_callback_uris`` too. I'd leave the check there since that function is also called in ``_on_workload_container_pebble_ready``.